### PR TITLE
[IO-625] FileUtils.copyDirectoryToDirectory does not reflect srcDir in exception message when srcDir is not a directory

### PR DESCRIPTION
**Pull request by mikkomaunu**
Fix and unit tests for https://issues.apache.org/jira/browse/IO-625.

### CODE REVIEW

File: src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * This class ensure the correctness of {@link FileUtils#copyDirectoryToDirectory(File, File)} (File,File)}.
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java]: I think the `@link` formatting is messed up here.

File: src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ * @see FileUtils#copyDirectoryToDirectory(File, File)
+ */
+public class FileUtilsCopyDirectoryToDirectoryTestCase {
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Test
+    public void copyDirectoryToDirectoryThrowsIllegalExceptionWithCorrectMessageWhenSrcDirIsNotDirectory() throws IOException {
+        File srcDir = temporaryFolder.newFile("notadirectory");
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java]: Use `final` where possible.

File: src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        try {
+            FileUtils.copyDirectoryToDirectory(srcDir, destDir);
+        } catch (Exception e) {
+            String msg = e.getMessage();
+            assertEquals(expectedExceptionType, e.getClass());
+            assertEquals(expectedMessage, msg);
+            return;
+        }
+        fail();
+
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java]: Remove extra line.

File: src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java (hunk 2 above)
Comment (mikkomaunu) [reply src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java]: Fixed in https://github.com/apache/commons-io/pull/95/commits/a3a0bfa102f7b304c067d670a7bf32aa45863a6e

File: src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java (hunk 1 above)
Comment (mikkomaunu) [reply src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java]: Fixed in https://github.com/apache/commons-io/pull/95/commits/a3a0bfa102f7b304c067d670a7bf32aa45863a6e

File: src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java (hunk 3 above)
Comment (mikkomaunu) [reply src/test/java/org/apache/commons/io/FileUtilsCopyDirectoryToDirectoryTestCase.java]: Fixed in https://github.com/apache/commons-io/pull/95/commits/a3a0bfa102f7b304c067d670a7bf32aa45863a6e

**Comment by garydgregory** [comment]
Looks good :-)
One more thing, (sorry) please add an entry to `src/changes/changes.xml`.

**Pull request merged**